### ISSUE TITLE
Classpath issues with adding Giter8

### ISSUE THREAD

**Issue by dwijnand** [sbt/sbt#2761]
1) sbt-android trying to use Javassist, by @pfn
- https://gitter.im/sbt/sbt?at=57ec2e83783fe9b945a59741

2) sbt-scalajs needing to use commons-compress 1.9 but failing because it gets commons-compress 1.6, by @dispalt
- https://gitter.im/sbt/sbt?at=57e9b2caa96e034d3d9a29c5
- https://gitter.im/sbt/sbt?at=57ea9f60d02791686c3f8c82

3) Older versions of sbt-git using older version of JGit and failing, by @SethTisue
- https://github.com/scala/community-builds/pull/290#issuecomment-248458277

**Comment by dwijnand** [comment]
This is where Giter8 was added: https://github.com/sbt/sbt/pull/2705

**Comment by dwijnand** [comment]
It seems that the versions of the transitive dependencies of Giter8 always win over any other version of the same dependency, which is why it's using a newer version of JGit than old sbt-git can work with and an older version of commons-compress than current sbt-android can work with.

@retronym suggests forking instead, and the fact that we're already going through a Java `sbt.template.TemplateResolver` interface (https://github.com/sbt/template-resolver) puts us in a good position to do so (hat tip to both him for the forking tip and @eed3si9n for making sbt/template-resolver a Java interface).

**Comment by eed3si9n** [comment]
Or we can send a PR to Giter8 to shade its dependencies.

**Comment by dwijnand** [comment]
On it

**Comment by dwijnand** [comment]
@eed3si9n before I go down the wrong rabbit hole, this means using sbt-assembly and publishing an assembly jar, right?

**Comment by pfn** [comment]
```
$ mkdir sbt2761
$ cd sbt2761
$ mkdir project
$ echo "sbt.version=0.13.13-RC2" > project/build.properties
$ echo 'addSbtPlugin("org.scala-android" % "sbt-android" % "1.6.18")' > project/plugins.sbt
$ sbt "gen-android com.foo Foo" android:ndkJavah
Getting org.scala-sbt sbt 0.13.13-RC2 ...
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/sbt/0.13.13-RC2/jars/sbt.jar ...
... snip ...
Downloaded https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/ivy/0.13.13-RC2/jars/ivy.jar
[info] Fetched artifacts of sbt2761-build
[info] Set current project to sbt2761 (in build file:/C:/Users/pfnguyen/src/sbt2761/)
[info] Creating project: Foo
Computing updates...                                                       100%
[info] Creating SBT project files
[warn] SBT files already exist in project/, not creating project/android.sbt
[info] Loading global plugins from C:\Users\pfnguyen\.sbt\0.13\plugins
[info] Loading project definition from C:\Users\pfnguyen\src\sbt2761\project
[info] Updating sbt2761-build
Downloading https://repo.typesafe.com/typesafe/ivy-releases/org.codehaus.plexus/plexus/1.0.4/ivys/ivy.xml.sha1
Downloading https://repo.typesafe.com/typesafe/ivy-releases/org.codehaus.plexus/plexus/1.0.4/ivys/ivy.xml
Downloading https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.codehaus.plexus/plexus/1.0.4/ivys/ivy.xml
Downloading https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.codehaus.plexus/plexus/1.0.4/ivys/ivy.xml.sha1
[info] Resolved sbt2761-build dependencies
[info] Fetching artifacts of sbt2761-build
[info] Fetched artifacts of sbt2761-build
[info] Set current project to sbt2761 (in build file:/C:/Users/pfnguyen/src/sbt2761/)
[info] Updating sbt2761
[info] Resolved sbt2761 dependencies
[info] Fetching artifacts of sbt2761
[info] Fetched artifacts of sbt2761
[info] Unpacking aar: support-vector-drawable-24.0.0.aar to com.android.support-support-vector-drawable-24.0.0
[info] Unpacking aar: animated-vector-drawable-24.0.0.aar to com.android.support-animated-vector-drawable-24.0.0
[info] Unpacking aar: appcompat-v7-24.0.0.aar to com.android.support-appcompat-v7-24.0.0
[info] Unpacking aar: support-v4-24.0.0.aar to com.android.support-support-v4-24.0.0
[info] Collecting resources
[info] Performing full resource merge
[info] Processing resources
[info] Regenerating TR.scala because R.java has changed
[info] Rebuilding all classes because R.java has changed
[info] Compiling 3 Scala sources and 3 Java sources to C:\Users\pfnguyen\src\sbt2761\target\android\intermediates\classes...
[info] 'compiler-interface' not yet compiled for Scala 2.11.8. Compiling...
[info]   Compilation completed in 8.022 s
[info] Packaging C:\Users\pfnguyen\src\sbt2761\target\android\intermediates\classes.jar ...
[info] Done packaging.
java.lang.RuntimeException: by java.lang.NoClassDefFoundError: javassist/util/proxy/ProxyObject
        at javassist.util.proxy.ProxyFactory.createClass3(ProxyFactory.java:515)

        at javassist.util.proxy.ProxyFactory.createClass2(ProxyFactory.java:492)

        at javassist.util.proxy.ProxyFactory.createClass1(ProxyFactory.java:428)

        at javassist.util.proxy.ProxyFactory.createClass(ProxyFactory.java:400)
        at javassist.util.proxy.ProxyFactory.create(ProxyFactory.java:675)
        at javassist.util.proxy.ProxyFactory.create(ProxyFactory.java:660)
        at android.NativeFinder$.apply(NativeFinder.scala:40)
        at android.Tasks$$anonfun$31.apply(tasks.scala:406)
        at android.Tasks$$anonfun$31.apply(tasks.scala:405)
        at scala.Function6$$anonfun$tupled$1.apply(Function6.scala:35)
        at scala.Function6$$anonfun$tupled$1.apply(Function6.scala:34)
        at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
        at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:237)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: javassist.CannotCompileException: by java.lang.NoClassDefFoundError: javassist/util/proxy/ProxyObject
        at javassist.util.proxy.FactoryHelper.toClass(FactoryHelper.java:170)
        at javassist.util.proxy.ProxyFactory.createClass3(ProxyFactory.java:507)

        at javassist.util.proxy.ProxyFactory.createClass2(ProxyFactory.java:492)

        at javassist.util.proxy.ProxyFactory.createClass1(ProxyFactory.java:428)

        at javassist.util.proxy.ProxyFactory.createClass(ProxyFactory.java:400)
        at javassist.util.proxy.ProxyFactory.create(ProxyFactory.java:675)
        at javassist.util.proxy.ProxyFactory.create(ProxyFactory.java:660)
        at android.NativeFinder$.apply(NativeFinder.scala:40)
        at android.Tasks$$anonfun$31.apply(tasks.scala:406)
        at android.Tasks$$anonfun$31.apply(tasks.scala:405)
        at scala.Function6$$anonfun$tupled$1.apply(Function6.scala:35)
        at scala.Function6$$anonfun$tupled$1.apply(Function6.scala:34)
        at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
        at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:237)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoClassDefFoundError: javassist/util/proxy/ProxyObject
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:760)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at javassist.util.proxy.FactoryHelper.toClass2(FactoryHelper.java:182)
        at javassist.util.proxy.FactoryHelper.toClass(FactoryHelper.java:164)
        at javassist.util.proxy.ProxyFactory.createClass3(ProxyFactory.java:507)

        at javassist.util.proxy.ProxyFactory.createClass2(ProxyFactory.java:492)

        at javassist.util.proxy.ProxyFactory.createClass1(ProxyFactory.java:428)

        at javassist.util.proxy.ProxyFactory.createClass(ProxyFactory.java:400)
        at javassist.util.proxy.ProxyFactory.create(ProxyFactory.java:675)
        at javassist.util.proxy.ProxyFactory.create(ProxyFactory.java:660)
        at android.NativeFinder$.apply(NativeFinder.scala:40)
        at android.Tasks$$anonfun$31.apply(tasks.scala:406)
        at android.Tasks$$anonfun$31.apply(tasks.scala:405)
        at scala.Function6$$anonfun$tupled$1.apply(Function6.scala:35)
        at scala.Function6$$anonfun$tupled$1.apply(Function6.scala:34)
        at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
        at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:237)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: javassist.util.proxy.ProxyObject
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:760)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at javassist.util.proxy.FactoryHelper.toClass2(FactoryHelper.java:182)
        at javassist.util.proxy.FactoryHelper.toClass(FactoryHelper.java:164)
        at javassist.util.proxy.ProxyFactory.createClass3(ProxyFactory.java:507)

        at javassist.util.proxy.ProxyFactory.createClass2(ProxyFactory.java:492)

        at javassist.util.proxy.ProxyFactory.createClass1(ProxyFactory.java:428)

        at javassist.util.proxy.ProxyFactory.createClass(ProxyFactory.java:400)
        at javassist.util.proxy.ProxyFactory.create(ProxyFactory.java:675)
        at javassist.util.proxy.ProxyFactory.create(ProxyFactory.java:660)
        at android.NativeFinder$.apply(NativeFinder.scala:40)
        at android.Tasks$$anonfun$31.apply(tasks.scala:406)
        at android.Tasks$$anonfun$31.apply(tasks.scala:405)
        at scala.Function6$$anonfun$tupled$1.apply(Function6.scala:35)
        at scala.Function6$$anonfun$tupled$1.apply(Function6.scala:34)
        at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
        at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:237)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
[error] (android:ndkJavah) by java.lang.NoClassDefFoundError: javassist/util/proxy/ProxyObject
[error] Total time: 18 s, completed Sep 29, 2016 8:56:37 AM
```

**Comment by eed3si9n** [comment]
@dwijnand Probably.

**Comment by dwijnand** [comment]
Fixed with #2780. We'll forward port this in 1.0.0.
